### PR TITLE
Contar

### PR DESCRIPTION
**Pull request by castorinop**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### cont.ar site extractor

Can extract info for get single video, complete serie, full channel and anothers filters like by genre

### CODE REVIEW

File: youtube_dl/extractor/contar.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+                entries.append(info)
+
+        return self.playlist_result(
+            entries, serie_id,
+            serie_info.get('name'), serie_info.get('story_large'))
+
+
+class ContarChannelIE(ContarBaseIE):
+
+    _UUID_RE = r'[\d]{1,}'
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: ?

File: youtube_dl/extractor/contar.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        'url': 'https://www.cont.ar/channel/242',
+        'info_dict': {
+            'id': '242',
+            'title': 'PAKAPAKA',
+            'description': '\xa0PAKAPAKA'
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: Do.

File: youtube_dl/extractor/contar.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    _TEST = {
+        'url': 'https://www.cont.ar/browse/genre/46',
+        'info_dict': {
+            'id': '46',
+            'title': 'Infantil',
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: Do.

File: youtube_dl/extractor/contar.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+            'season_number': 1,
+            'episode_number': 1,
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
+        },
+        'params': {
+            'usenetrc': True,
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: No.

File: youtube_dl/extractor/contar.py [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+            'duration': 648,
+            'release_year': 2016,
+            'description': 'Matilde llega a la casa de su abuelo Barbacrespa y en ella accidentalmente rompe un coco m\u00e1gico que la dota de poder.',
+            'season_number': 1,
+            'episode_number': 1,
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: Do.

File: youtube_dl/extractor/contar.py [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+        video_id = self._match_id(url)
+
+        video = self._call_api('videos/' + video_id, video_id, headers={'Referer': url})
+        info = self._get_video_info(video, video_id)
+        return info
+
+
+class ContarSerieIE(ContarBaseIE):
+
+    _UUID_RE = r'[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}'
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: DRY.

File: youtube_dl/extractor/contar.py [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+        'info_dict': {
+            'id': '353247d5-da97-4cb6-8571-c4fbab28c643',
+            'title': 'Vidas de Radio',
+            'description': 'Ana Gerschenson conduce el ciclo que repasa historias de grandes personalidades que le dieron vida al medio; marcaron una época de la Argentina y de tu vida, esas voces amigas que estuvieron siempre y son Vidas De Radio.'
+            # 'thumbnail': r're:^https?://.*\.jpg$',
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: Do.

File: youtube_dl/extractor/contar.py (hunk 4 above)
Comment (castorinop) [reply youtube_dl/extractor/contar.py]: what is the correct way to pass username and password to run the test?

Comment (dstftw) [reply youtube_dl/extractor/contar.py]: By providing username and password in params obviously.

File: youtube_dl/extractor/contar.py [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+                entries.append(info)
+
+        return self.playlist_result(
+            entries, serie_id,
+            serie_info.get('name'), serie_info.get('story_large'))
+
+
+class ContarChannelIE(ContarBaseIE):
+
+    _UUID_RE = r'\d+'
```
Comment (dstftw) [inline youtube_dl/extractor/contar.py]: Nothing changed.

File: youtube_dl/extractor/contar.py (hunk 6 above)
Comment (castorinop) [reply youtube_dl/extractor/contar.py]: Can you explain to me more what is the observation?

Comment (dstftw) [reply youtube_dl/extractor/contar.py]: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

File: youtube_dl/extractor/contar.py (hunk 8 above)
Comment (castorinop) [reply youtube_dl/extractor/contar.py]: Should I embed it in _VALID_URL?

File: youtube_dl/extractor/contar.py (hunk 6 above)
Comment (castorinop) [reply youtube_dl/extractor/contar.py]: moving _UUID_RE to ContarBaseIE and call from ContarIE and ContarSerieIE is the better solution?

**Comment by castorinop** [comment]
Can you update review ?

**Pull request closed without merging**